src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -91,53 +80,6 @@ class Decimal extends React.Component {
     );
   };
 
-  onPaste = (ev) => {
-    const value = this.replace(ev);
-    const isValid = this.isValidDecimal(value);
-    if (!isValid) {
-      ev.preventDefault();
-    }
-  };
-
-  onKeyPress = (ev) => {
-    // We're checking to see if the proposed change is valid, if not we block the user input
-    if (!this.isValidKeyPress(ev)) {
-      ev.preventDefault();
-    }
-    if (this.props.onKeyPress) {
-      this.props.onKeyPress(ev);
-    }
-  };
-
-  isValidKeyPress = (ev) => {
-    const { delimiter, separator } = I18nHelper.format();
-
-    if (
-      Events.isNumberKey(ev) ||
-      ev.key === delimiter ||
-      ev.key === separator ||
-      Events.isDeletingKey(ev)
-    ) {
-      return this.isValidDecimal(this.replace(ev));
-    }
-
-    if (ev.ctrlKey || ev.metaKey) {
-      // user is doing some browser related task, we don't want to interfere with that.
-      // Command is metaKey in Safari, but ctrlKey in all other browsers
-      // Windows key is metaKey also
-      // we do want to check paste, we do that onPaste
-      return true;
-    }
-
-    if (Events.isMinusKey(ev) && ev.target.selectionStart === 0) {
-      // user is entering a negative symbol at the start of the number
-      return true;
-    }
-
-    this.startAnimation();
-    return false;
-  };
-
   onBlur = () => {
```

Comment:
Won't the logic in here need to update as well?

Reply:
I think we can remove all of the `onBlur` code because it is only handling the `-` key press. If a user pressed `-` accidentally instead of `9` we could pick up the user error in validations because we're reverting to the `defaultValue` currently.